python/ray/tests/test_asyncio.py

```diff
@@ -198,6 +198,26 @@ async def test_asyncio_double_await(ray_start_regular_shared):
     await waiting
 
 
+@pytest.mark.asyncio
+async def test_asyncio_exit_actor(ray_start_regular_shared):
+    # https://github.com/ray-project/ray/issues/12649
+    # The test should just hang without the fix.
+
+    @ray.remote
+    class Actor:
+        async def exit(self):
+            ray.actor.exit_actor()
+
+        async def ping(self):
+            return "pong"
```

Comment:
We should modify this test (or add an additional one) to make this method an infinite loop. That verifies the behavior that exit_actor() exits immediately, not once all tasks are completed.